fairseq/training.py

```diff
@@ -0,0 +1,434 @@
+                            i += 1
+
+        colors = ["#2a4d6e", "#aa3c39"]
+        import matplotlib.pyplot as plt
+
+        fig, axs = plt.subplots(2, sharex=True, figsize=(12, 9))
+        train_epochs = list(train_score.keys())
+        valid_epochs = list(valid_score.keys())
+        lr_epochs = list(learning_rates.keys())
+        train_epochs.sort()
+        valid_epochs.sort()
+        lr_epochs.sort()
+
+        axs[0].plot(
+            train_epochs,
+            [train_score[e] for e in train_epochs],
+            "o-",
+            color=colors[0],
+            label="train score",
+        )
+        axs[0].plot(
+            valid_epochs,
+            [valid_score[e] for e in valid_epochs],
+            "o-",
+            color=colors[1],
+            label="valid score",
+        )
+        axs[0].set_ylabel("score")
+        axs[0].legend()
+
+        axs[1].plot(
+            train_epochs,
+            [train_error[e] for e in train_epochs],
+            "o-",
+            color=colors[0],
+            label="train error",
+        )
+        axs[1].plot(
+            valid_epochs,
+            [valid_error[e] for e in valid_epochs],
+            "o-",
+            color=colors[1],
+            label="valid error",
+        )
+        axs[1].set_ylabel("error")
+        axs[1].set_xlabel("epochs")
+        axs[1].legend()
+
+        plt.savefig(self.out_plot_se)
+
+        fig, axs = plt.subplots()
+        axs.plot(
+            lr_epochs,
+            [learning_rates[e] for e in lr_epochs],
+            "o-",
+            color=colors[0],
+            label="learning rate",
+        )
+        axs.set_ylabel("learning rate")
+        axs.set_xlabel("epochs")
+        axs.legend()
+
+        plt.savefig(self.out_plot_lr)
+
+    def _get_run_cmd(self):
+        run_cmd = [
+            "--config-dir",
+            os.path.dirname(self.out_fairseq_hydra_yaml.get_path()),
+            "--config-name",
+            os.path.basename(self.out_fairseq_hydra_yaml.get_path()),
+        ]
+        run_cmd += self.command_line_args
+        run_cmd += ["checkpoint.save_dir=" + self.out_checkpoint_dir.get_path()]
+        if self.save_interval != self.fairseq_hydra_config.config_dict.get(
+            "checkpoint", {}
+        ).get("save_interval", None):
+            run_cmd += ["checkpoint.save_interval=" + str(self.save_interval)]
+        if self.max_epoch != self.fairseq_hydra_config.config_dict.get(
+            "optimization", {}
+        ).get("max_epoch", None):
+            run_cmd += ["optimization.max_epoch=" + str(self.max_epoch)]
+
+        if self.use_cache_manager:
+            run_cmd += ["task.data=" + self.out_cached_audio_manifest.get_path()]
+
+        if self.fairseq_root is not None:
+            sys.path.insert(0, self.fairseq_root)
+            hydra_train_entry = self.fairseq_root + "fairseq_cli/hydra_train.py"
+            run_cmd.insert(0, tk.uncached_path(hydra_train_entry))
```

Comment:
shun `tk.uncached_path`. Once you decide to only use `tk.Path` you can get `get_path()` to convert to `str`